src/uct/cuda/cuda_copy/cuda_copy_ep.c

```diff
@@ -133,10 +130,14 @@ ucs_status_t uct_cuda_copy_ep_put_short(uct_ep_h tl_ep, const void *buffer,
                                         unsigned length, uint64_t remote_addr,
                                         uct_rkey_t rkey)
 {
+    uct_cuda_copy_iface_t *iface = ucs_derived_of(tl_ep->iface, uct_cuda_copy_iface_t);
     ucs_status_t status;
 
-    status = UCT_CUDA_FUNC(cudaMemcpy((void *)remote_addr, buffer,
-                                      length, cudaMemcpyHostToDevice));
+    UCT_CUDA_COPY_CHECK_AND_CREATE_STREAM(iface->stream_h2d);
+
+    UCT_CUDA_FUNC(cudaMemcpyAsync((void *)remote_addr, buffer, length,
+                                  cudaMemcpyHostToDevice, iface->stream_h2d));
+    status = UCT_CUDA_FUNC(cudaStreamSynchronize(iface->stream_h2d));
```

Comment:
Do we need to synchronize here or can we defer to a later point to synchronize multiple operations?

Reply:
@Akshay-Venkatesh I don't think that would be easy.  At least this PR is not targeting for this.

Reply:
This is good for now given that there's a non blocking stream used.